zign-sec/src/main/java/com/hedvig/external/zignSec/ZignSecSessionServiceImpl.kt

```diff
@@ -293,23 +300,73 @@ class ZignSecSessionServiceImpl(
         ZignSecBankIdProgressStatus.FAILED
     }
 
-    companion object {
-        private val logger = LoggerFactory.getLogger(ZignSecServiceImpl::class.java)
+    fun validatePersonNumberAgainstDateOfBirth(personNumber: String, dateOfBirth: String, method: ZignSecAuthenticationMethod): Boolean {
+        if (personNumber.dayMonthAndTwoDigitYearFromNorwegianOrDanishSsn() == dateOfBirth.dayMonthAndTwoDigitYearFromDateOfBirth()) {
+            return true
+        }
+
+        return when (method) {
+            ZignSecAuthenticationMethod.NORWAY_WEB_OR_MOBILE ->
+                validatePersonNumberAgainstDateOfBirthWithKnownNorwegianBankIdBug(personNumber, dateOfBirth) ||
```

Comment:
Unsure where to put the suggestion, but working with LocalDates would make this a lot easier. Especially since "plusDays" and "minusDays" will take into account if the day is the last of the month.
```suggestion
        val birthDate = LocalDate....from(dateOfBirth)
        val ssnBirthDate = LocalDate....from(personNumber)
        
        return when (method) {
            ZignSecAuthenticationMethod.NORWAY_WEB_OR_MOBILE ->
                !birthDate.isBefore(ssnBirthDate.minusDays(4) && !birthDate.isAfter(ssnBirthDate.plusDays(4))
```

Reply:
+1. The `java.time` package is amazing and the `Local*` types are perfect for doing date algebra like this.

Reply:
I chose to not use Dates because of 2 reasons 1. We don't actually have the year in the ssn 2. It will not work with when it is actually not a date. Like 33:th of January.